test/unit/Reflection/TypeGetMethodRecursivelyExtensionsTests.cs

```diff
@@ -0,0 +1,123 @@
+using System;
+using System.Reflection;
+using RapidCore.Reflection;
+using Xunit;
+
+namespace RapidCore.UnitTests.Reflection
+{
+    public class TypeGetMethodRecursivelyExtensionsTests
+    {
+        [Fact]
+        public void GetMethodRecursively_FindsMethodsWithParams_InFirstLayer()
+        {
+            var actual = typeof(GuineaPig).GetMethodRecursively("OneParam", typeof(string));
+            
+            Assert.NotNull(actual);
+            Assert.Equal("OneParam", actual.Name);
+        }
+
+        [Fact]
+        public void GetMethodRecursively_FindsMethodsWithoutParams_InFirstLayer()
+        {
+            var actual = typeof(GuineaPig).GetMethodRecursively("ZeroParams");
+            
+            Assert.NotNull(actual);
+            Assert.Equal("ZeroParams", actual.Name);
+        }
+
+        [Fact]
+        public void GetMethodRecursively_FindsOverloadedMethodsWithoutParams()
+        {
+            var actual = typeof(GuineaPig).GetMethodRecursively("HasOverload");
+            
+            Assert.NotNull(actual);
+            Assert.Equal("HasOverload", actual.Name);
+            Assert.Empty(actual.GetParameters());
+        }
+
+        [Fact]
+        public void GetMethodRecursively_FindsOverloadedMethodsWithParams()
+        {
+            var actual = typeof(GuineaPig).GetMethodRecursively("HasOverload", typeof(Attribute));
+            
+            Assert.NotNull(actual);
+            Assert.Equal("HasOverload", actual.Name);
+            Assert.Equal(1, actual.GetParameters().Length);
+        }
+
+        [Fact]
+        public void GetMethodRecursively_FindsMethodsWithoutParams_InNextLayer()
+        {
+            var actual = typeof(GuineaPig).GetMethodRecursively("GetType");
+            
+            Assert.NotNull(actual);
+            Assert.Equal("GetType", actual.Name);
+            Assert.Equal(typeof(Object), actual.DeclaringType);
+        }
+
+        [Fact]
+        public void GetMethodRecursively_Throws_ifMethodDoesNotExist_withoutParams()
+        {
+            try
```

Comment:
Rather than try/catching when you are testing a method that is supposed to throw an exception, you can use the nice xunit Assert.Throws<> method:
http://hadihariri.com/2008/10/17/testing-exceptions-with-xunit/